### ISSUE TITLE
a question about Approximating the gradient field

### ISSUE THREAD

**Issue by lihongyi1234** [mkazhdan/PoissonRecon#242]
![image](https://user-images.githubusercontent.com/36233282/204531906-46346eb7-602e-4cc2-99cd-470a75001e3f.png)
why convert a integral to constant?

**Comment by mkazhdan** [comment]
Not sure I quite follow the question. But the point is that we don't know \vec{N} at arbitrary locations (and we don't actually know P_s since we don't have the surface).

**Comment by lihongyi1234** [comment]
![image](https://user-images.githubusercontent.com/36233282/206663738-e863f46d-4f47-4751-aa81-a5c7c92870ef.png)
hello，i know box filter, but i don't know n-th convolution, what does n-th convolution mean?

**Comment by mkazhdan** [comment]
Filter the box filter with the box filter.

**Comment by besthwang** [comment]
Does convolution looks like the following picuture in 1D.
 
![image](https://user-images.githubusercontent.com/49426029/231961383-dd599990-818d-4662-a80f-6f90995836d1.png)

https://i.stack.imgur.com/JTnjz.jpg

**Comment by mkazhdan** [comment]
Yes, though x_2 should be the same as x_1. (In this example, x_2 is half the width of x_1.)

You man find the discussion about B-splines at https://www.cs.jhu.edu/~misha/Notes/BSplines.html helpful.

**Comment by besthwang** [comment]
I found below simple program compute can quadratic b-splines

the result is following:
![image](https://user-images.githubusercontent.com/49426029/232364327-abf25ea1-afcb-4925-8a08-c8b5ca8b1fc9.png)



#include <pcl/PCLPointCloud2.h>
#include <pcl/io/pcd_io.h>
#include <pcl/io/vtk_io.h>
#include <pcl/console/print.h>
#include <pcl/console/parse.h>
#include <pcl/console/time.h>
#include <pcl/surface/poisson.h>
#include <pcl/surface/3rdparty/poisson4/bspline_data.h>

using namespace pcl;
using namespace pcl::poisson;
using namespace pcl::io;
using namespace pcl::console;

int main(int argc, char** argv) {
  PPolynomial<2> f1 = PPolynomial<2>::BSpline();
  f1.printnl();
  return 0;
}

**Comment by besthwang** [comment]
I saw BSplineComponet(int i) in Polynominal.ini 
and I found a equation:

<img width="517" alt="image" src="https://user-images.githubusercontent.com/49426029/232935915-e1db4bf5-a8dc-4721-8397-f45222222dab.png">

1. Could  you tell me what  reference the equation is generated. 


  const int Degree = 2;
  BSplineData<Degree, float>::BSplineComponents baseBSpline;
  int i = 2;
  Polynomial<Degree> p = Polynomial< Degree>::BSplineComponent(i);
  baseBSpline[i] = p.shift(double(-(Degree + 1) / 2) + i - 0.5);

2. also, I found above code 
   shift() is called,  dth bspline base is geneated for each i.
   
   
![image](https://user-images.githubusercontent.com/49426029/232937350-e407429a-81d1-48a2-8371-339725145df0.png)

   
    Could you tell me what algorithm or refercen is used here.
    
  Thanks.
